lib/site_prism/element_container.rb

```diff
@@ -228,14 +228,37 @@ def deduce_iframe_element_find_args(args)
     end
 
     def extract_section_options(args, &block)
-      if args.first.is_a?(Class)
-        section_class = args.shift
-      elsif block_given?
-        section_class = Class.new(SitePrism::Section, &block)
+      section_class = deduce_section_class(args, &block)
+      arguments = deduce_search_arguments(section_class, args)
+      [section_class, arguments]
+    end
+
+    def deduce_section_class(args, &block)
+      if args.first.is_a?(Class) && args.first.ancestors.include?(SitePrism::Section)
```

Comment:
Apologies if I'm missing something here, but this condition isn't always successful, even if a block is given.

This means that `base_class` can sometimes be an undefined local variable right?

Reply:
yes, you are right, in this case, the interpreter will treat it as `nil` [on line 242](https://github.com/natritmeyer/site_prism/pull/250/files#diff-0142fe7beaa583d4ce8030ab2a152817R242) and it will use SitePrism::Section as a base class

Reply:
Qu) When I do `var1 || 'string'` inside a ruby interpreter it crashes. Are you sure this will work? Aren't only instance variables `nil` by default? Or am I missing a further point?

Reply:
Sorry, I told you part of the truth...
```ruby
var1 || 'string' # Raises NameError
```

```ruby
if false # never can be reached
  var1 = 1
end

var1 || 'string' # Works. It's a part of how the ruby interpreter works. =(
```